Python/import.c

```diff
@@ -885,7 +869,25 @@ PyObject *
 PyImport_AddModuleObject(PyObject *name)
 {
     PyThreadState *tstate = _PyThreadState_GET();
-    return import_add_module(tstate, name);
+    PyObject *mod = import_add_module(tstate, name);
+    if (mod) {
+        if (Py_REFCNT(mod) == 1) {
+            /* This check does not preven an undefined behavior in the
+             * following code, because the module can have references
+             * to itself. */
+            PyErr_SetString(PyExc_RuntimeError, "Unexpected zero reference count");
+            Py_DECREF(mod);
+            return NULL;
+        }
+        Py_DECREF(mod);
+        if (Py_REFCNT(mod) == 0) {
```

Comment:
Maybe is overkill, but we could get a weakreference and check if is alive

Reply:
Good idea about weak ref. And I agree with your concern.